test/Dialect/ESI/manifest.mlir

```diff
@@ -18,8 +34,22 @@ hw.module @top(in %clk: !seq.clock, in %rst: i1) {
 
   %3 = esi.null : !esi.channel<!alias>
   esi.cosim %clk, %rst, %3, "t3" : !esi.channel<!alias> -> !esi.channel<i32>
+
+  esi.service.instance #esi.appid<"cosim"> svc @HostComms impl as "cosim" (%clk, %rst) : (!seq.clock, i1) -> ()
+  hw.instance "m1" @Loopback (clk: %clk: !seq.clock) -> () {esi.appid=#esi.appid<"loopback_inst">}
```

Comment:
Since you're walking and enumerating the instance hierarchy, it might be good to add a test which has multiple instantiations of the same module.